### PR TITLE
Clean up `src/NemoStuff.jl`

### PR DESCRIPTION
**Pull request by joschmitt**
Maybe some (more) of these functions could/should just go...

I'll try to add tests once the code coverage is in as far as this is possible in AbstractAlgebra.

### CODE REVIEW

File: src/Deprecations.jl [hunk 1]
```diff
@@ -54,3 +54,8 @@ import .Generic: exponent; @deprecate exponent(a::Generic.MPoly{T}, i::Int, j::I
 import .Generic: set_exponent_vector!; @deprecate set_exponent_vector!(a::Generic.MPoly{T}, i::Int, exps::Vector{Int}, ::Type{Val{ord}}) where {T <: RingElement, ord} set_exponent_vector!(a, i, exps, Val(ord))
 import .Generic: is_gen; @deprecate is_gen(x::Generic.MPoly{T}, ::Type{Val{ord}}) where {T <: RingElement, ord} is_gen(x, Val(ord))
 import .Generic: degree; @deprecate degree(f::Generic.MPoly{T}, i::Int, ::Type{Val{ord}}) where {T <: RingElement, ord} degree(f, i, Val(ord))
+
+# deprecated during 0.42.*
+@deprecate zero_matrix(::Type{MatElem}, R::Ring, n::Int, m::Int) zero_matrix(R, n, m)
+@deprecate identity_matrix(::Type{MatElem}, R::Ring, n::Int) zero_matrix(R, n)
```
Comment (lgoettgens) [inline src/Deprecations.jl:60]: iirc I talked to @thofma in the past about removing these two and he had some use-case for them

File: src/julia/Integer.jl [hunk 2]
```diff
@@ -28,6 +28,8 @@ is_exact_type(::Type{T}) where T <: Integer = true
 
 is_domain_type(::Type{T}) where T <: Integer = true
 
+base_ring(::Vector{T}) where T <: Integer = T
```
Comment (lgoettgens) [inline src/julia/Integer.jl:31]: This seems not what `base_ring` is supposed to do. I would be in favor of removing this if it does not break downstsream

Comment (joschmitt) [reply src/julia/Integer.jl:31]: It is used in Hecke. There is some code that should work for vectors of integers both as `Vector{Int}` and `ZZMatrix`. Maybe we should introduce `_base_ring` for this, I don't know.

File: src/Deprecations.jl (hunk 1 above)
Comment (thofma) [reply src/Deprecations.jl:60]: yes, don't remove this

Comment (joschmitt) [reply src/Deprecations.jl]: :+1: I put it in `src/Matrix.jl`.

File: src/julia/Integer.jl (hunk 2 above)
Comment (fingolfin) [reply src/julia/Integer.jl:31]: What is it used for? And why is it just for integer vectors? E.g. this gives an error:
```
julia> R,t = QQ[:t] ; base_ring([t])
ERROR: MethodError: no method matching base_ring(::Vector{QQPolyRingElem})
The function `base_ring` exists, but no method is defined for this combination of argument types.
```

This makes it even stranger that `base_ring` exists for e.g. `Vector{Int}`. Really seems to me that Hecke should use a differently named helper here?

Comment (fingolfin) [reply src/julia/Integer.jl:31]: If at all I'd expect `base_ring` on a vector to return the parent of the elements of the vector, which would suggest
```suggestion
base_ring(v::Vector{T}) where T = parent(v[1])
```
or if one just wants the type, then perhaps
```suggestion
base_ring_type(v::Vector{T}) where T = parent_type(T)
```

Comment (joschmitt) [reply src/julia/Integer.jl:31]: @thofma This is used in the Plesken--Souvignier lattice stuff. What do you think?

**Pull request merged**